File: dynamic_programming/118_pascals_triangle.cc
#include <vector>

// 2024/01/12: 自己实现

std::vector<std::vector<int>> Generate(int num_rows) {
  std::vector<std::vector<int>> results;
  for (int i = 0; i < num_rows; ++i) {
    std::vector<int> result(i + 1, 1);
    for (int j = 1; j < i; ++j) {
      result[j] = results[i - 1][j] + results[i - 1][j - 1];
    }
    results.push_back(result);
  }
  return results;
}

File: dynamic_programming/118_pascals_triangle.py
from typing import List


def generate(num_rows: int) -> List[List[int]]:
    if num_rows == 0:
        return []

    res = []
    for i in range(1, num_rows + 1):
        layer = [1] * i
        for j in range(1, i - 1):
            layer[j] = res[-1][j - 1] + res[-1][j]
        res.append(layer)

    return res


num_rows = 1
print(generate(num_rows))
